### DISCUSSION TITLE
OMV and reverse proxy (caddy as docker container) and nexctcloud-aio (as docker container)

### DISCUSSION THREAD

**Discussion by ho16** [nextcloud/all-in-one#1509] [Questions] [answered]
Hello,

I have a single pc for handling the NAS stuff at home. 
It is working with openmediavault, where the docker environment is used to handle the nextcloud containers. 
I am using a dyndns service for getting a public IP and domain. The reverse proxy (caddy), running in a container on that machine is prepared to prepare certificates for the public domain by letsencrypt. 

When starting the nextcloud-aio container, my goal is to just direct the external traffic from the external request to it (what a reverse proxy normaly makes).

I have problems using the master container, it is failing in the domain validation. When trying to skip it, which the settinsg skip-domain-validation, it is starting, I am coming to the point, where the containers are started. But I could not connect to it. 

Does anyone have a similar infrastructure, which is working?

Thanks

**Comment by szaimen** [comment]
Hi, can you please post all `docker run` commands or all the docker-compose.yml files that you used here?

**Reply by ho16** [reply]
Hi szaimen,

I am doubting the setting with the network host option and with docker network nextcloud. 
Even, when both are using the docker network nextcloud I cannot login. 
I think that port 11000 Is this the problem, isn't it?
As docker is running on openmediavault some ports are already reserved, (443 and 8080), so I tried to map them different.

docker run for the master container:

----------------------------------------------

```
sudo docker run \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 58080:8080 \
-e APACHE_PORT=11000 \
-e APACHE_IP_BINDING=127.0.0.1 \
-e SKIP_DOMAIN_VALIDATION=true \
-e NEXTCLOUD_DATADIR=/srv/dev-by-disk-label-Data1/Docker/nextcloud-aoi/data \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
--network host \
nextcloud/all-in-one:latest
```

------------------------------------------------

This is the docker create for the caddy reverse proxy:

---------------------------------------------------

```
docker create \
--name caddy \
-p 50090:80 \
-p 50453:443 \
-v /srv/dev-disk-by-label-Data1/Docker/caddy/site:/srv \
-v /srv/dev-disk-by-label-Data1/Docker/caddy/data:/data \
-v /srv/dev-disk-by-label-Data1/Docker/caddy/config:/config \
-v /srv/dev-disk-by-label-Data1/Docker/caddy/Caddyfile:/etc/caddy/Caddyfile \
--network nextcloud \
caddy:latest 
```

----------------------------------------------

And the Caddyfile used in the above container:
--------------------------------------------------------

```
{
	#General Options
	debug
}

	<My dyndns sub domain> {
		#respond "Proxy Reached"
		header Strict-Transport-Security max-age=31536000;
		reverse_proxy  127.0.0.1:11000	
	}
```

--------------------------------------------------------

**Comment by szaimen** [comment]
You made a mistake: `--network host` needs to be applied onto the caddy container and not to the AIO container. That should make it work.

**Reply by ho16** [reply]
Hello, 
but this is not possible. Using the host mode, discards all port mapping. 
When doing so, I get the following message in the cl:
WARNING: Published ports are discarded when using host network mode

Port 443 and 80 are in use from Open Mediavault.

**Comment by szaimen** [comment, marked as answer]
I see. In that case are your options to run the reverse proxy in a VM that uses bridged network mode  (so that the reverse proyy gets its own internal ip-address and can use port 80 and 443 again) or install the reverse proxy on a different device in your home network or change the omv interface to not use port 80 and 443.

**Reply by ho16** [reply]
That's what I did in the docker create command. The network nextcloud is of type bridge. 
It is working. from the external access.
Can it be, that there is a problem with the reverse proxy command ?

**Reply by szaimen** [reply]
No, you did something different. But anyway, if you wanto to do it like this, you need to specify the ip-address of the server and not use 127.0.0.1 as target in your Caddyfile.

**Reply by szaimen** [reply]
You can find the ip-address with this command that you need to execute under omv: `ip a | grep "scope global" | head -1 | awk '{print $2}' | sed 's|/.*||'`

**Reply by ho16** [reply]
Ok, I found the ip address of the omv server with the command. Now I am using it in the caddy file.
I tried for all container to use the bridge network "nextcloud".

After a new try, the bextcloud -aoi installation is up with all rtequired containers. But connecting is still nto possible. 
Using the public domain is not working. Using the reverse proxy address with the ip addres and port 11000 is not possible as well. 

Any ideas?

**Reply by szaimen** [reply]
Ah yes, you need to remove `-e APACHE_IP_BINDING=127.0.0.1 \` from your AIO config now that the reverse proxy is not running on the host...

**Reply by szaimen** [reply]
Better is to set it to `0.0.0.0` instead of removing the value.

**Reply by ho16** [reply]
Great, setting the value in APACHE_IP_BINBDING to 0.0.0.0 fixed my issue. 
Now I was able to prepare the first user and the credentials for the database connection.
The access to it is working over my public dyndns domain.

Thank you very muc for the help,.
